pkg/operation/operation.go

```diff
@@ -349,6 +352,56 @@ func (o *Operation) ShootVersion() string {
 	return o.Shoot.Info.Spec.Kubernetes.Version
 }
 
+func (o *Operation) newTerraformer(purpose, namespace, name string) (*terraformer.Terraformer, error) {
+	image, err := o.ImageVector.FindImage(common.TerraformerImageName, o.K8sSeedClient.Version(), o.K8sSeedClient.Version())
+	if err != nil {
+		return nil, err
+	}
+
+	return terraformer.NewForConfig(o.Logger, o.K8sSeedClient.RESTConfig(), purpose, namespace, name, image.String())
+}
+
+// NewBackupInfrastructureTerraformer creates a new Terraformer for the matching BackupInfrastructure.
+func (o *Operation) NewBackupInfrastructureTerraformer() (*terraformer.Terraformer, error) {
+	var backupInfrastructureName string
+	if o.Shoot != nil {
+		backupInfrastructureName = common.GenerateBackupInfrastructureName(o.Shoot.SeedNamespace, o.Shoot.Info.Status.UID)
+	} else {
+		backupInfrastructureName = o.BackupInfrastructure.Name
+	}
+
+	return o.newTerraformer(common.TerraformerPurposeBackup, common.GenerateBackupNamespaceName(backupInfrastructureName), backupInfrastructureName)
+}
+
+// NewShootTerraformer creates a new Terraformer for the current shoot with the given purpose.
+func (o *Operation) NewShootTerraformer(purpose string) (*terraformer.Terraformer, error) {
+	return o.newTerraformer(purpose, o.Shoot.SeedNamespace, o.Shoot.Info.Name)
+}
+
+// ChartInitializer initializes a terraformer based on the given chart and values.
+func (o *Operation) ChartInitializer(chartName string, values map[string]interface{}) terraformer.Initializer {
```

Comment:
Why have we kept the `terraformer-common` chart?
It contains the `ServiceAccount`, the `RoleBinding`, and the `ConfigMap`s+`Secret`s.
The first two (`ServiceAccount`,`RoleBinding`) seem to be ensured during execution of Terraformer (https://github.com/adracus/gardener/blob/refactor.terraformer/pkg/operation/terraformer/terraformer.go#L271 and https://github.com/adracus/gardener/blob/refactor.terraformer/pkg/operation/terraformer/terraformer.go#L287), so it's not needed anymore.

Can we also get rid of the `ConfigMap`s and `Secret`s?
The logic for creating these resources without charts seems to be built-in already (https://github.com/adracus/gardener/blob/refactor.terraformer/pkg/operation/terraformer/terraformer.go#L287), so why not letting the users just provide the `main.tf`, `terraform.tfvars` and `variables.tf` directly instead of this chart here?

Then we could drop this function.

Reply:
I got rid of the `ServiceAccount` and `RoleBinding` templates because they were indeed not needed anymore.

Regarding the `ConfigMap`s and `Secret`s: Yes, the logic is built in already, but right now we have tons of charts that actually depend on this functionality as they render ther `main`, `tfvars` and `variables` with helm charts. Rewriting these to make use of the new initializer would take quite a lot of time IMHO. It can be done for sure, but this increases the size of this PR significantly + it requires rewriting of the charting logic to output the individual files, which is hard as well.

Reply:
OK, makes sense. Let's get rid of the `terraformer-common` chart as soon as we have moved out the provider specifics - then it should indeed be not needed anymore, right?

Reply:
Definitely, that was also the reason why we decoupled this old, char-based initializer into the operation class.